src/sentry/ownership/grammar.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import absolute_import
+
+from collections import namedtuple
+from parsimonious.grammar import Grammar, NodeVisitor
+
+__all__ = ('parse_rules', 'dump_schema', 'load_schema')
+
+VERSION = 1
+
+# Grammar is defined in EBNF syntax.
+ownership_grammar = Grammar(r"""
+
+ownership = line+
+
+line = _ (comment / rule / empty) newline?
+
+rule = _ matcher owners
+
+matcher      = _ matcher_tag identifier
+matcher_tag  = (matcher_type sep)?
+matcher_type = "url" / "path"
+
+owners       = _ owner+
+owner        = _ team_prefix identifier
+team_prefix  = "#"?
+
+comment = ~r"#[^\r\n]*"
+
+# TODO: make more specific
+identifier = ~r"\S+"
```

Comment:
What do you think this needs to be more specific about? IMO this is probably fine, since it matches up to a space right?

Reply:
Correct, but ultimately it'll match like, a team slug or a user email address, so I can restrict these a bit more instead of a generic "everything except a space".